Vault 46599/add missing issuer external id docs (#2787)
hashicorp/web-unified-docs: content/vault/v2.x/content/api-docs/secret/identity/entity-alias.mdx

- `canonical_id` `(string: <required>)` - Entity ID to which this alias belongs to.

- `mount_accessor` `(string: <required>)` - Accessor of the mount to which the
  alias should belong to. Can be omitted when `issuer` is set (see the note below).

- `custom_metadata` `(map<string|string>: <optional>)` - A map of arbitrary string to string valued 
  user-provided metadata meant to describe the alias.

- `issuer` `(string: <optional>)` - Issuer URL of the external identity provider
  bound to this alias. Can be provided instead of `mount_accessor`. Cannot be
  changed after creation.

- `external_id` `(string: <optional>)` - Subject identifier for the alias from the
  external identity provider. Must be unique with `issuer` within a namespace.
  Cannot be changed after creation.

~> **NOTE:** For aliases bound to an OAuth resource-server configuration
(Enterprise), `mount_accessor` is a synthetic value with the format
`oauth-resource-server_<namespace_id>_<config_id>`, where `<config_id>` is the

> UUID of the configuration. Provide `issuer` instead and Vault generates it for you.

Reviewer: Use the callout syntax: `<Note>` ([Style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/appendix.md#alert-types))

```suggestion
<Note>

For aliases bound to an OAuth resource-server configuration
(Enterprise), `mount_accessor` is a synthetic value with the format
`oauth-resource-server_<namespace_id>_<config_id>`, where `<config_id>` is the
UUID of the configuration. Provide `issuer` instead and Vault generates it for you.

</Note>
```

- `canonical_id` `(string: <required>)` - Entity ID to which this alias belongs to.

- `mount_accessor` `(string: <required>)` - Accessor of the mount to which the

>   alias should belong to. Can be omitted when `issuer` is set (see the note below).

Reviewer 2: ```suggestion
  alias should belong to. You can omit `mount_accessor` during creation when you
  set `issuer`. If you omit `mount_accessor`, Vault automatically generates a
  synthetic value for aliases bound to an OAuth resource-server configuration in
  the format `oauth-resource-server_<namespace_id>_<config_id>`, where `<config_id>`
  is the UUID of the matching configuration.
```
There's no need to move the information to a separate section when it's most relevant (and most likely to be seen) here.

Clarifying question: does "can be omitted" mean you can set both `mount_accessor` and `issuer` if you want? Or are they mutually exclusive?

Author: `mount_accessor` and `issuer` are not mutually exclusive—you can set both. Vault only auto-generates the synthetic mount_accessor when you provide `issuer` and leave `mount_accessor` empty; if you supply both, Vault uses the one you provide. I went ahead and made the changes you requested, let me know if it should be changed past that!

- `custom_metadata` `(map<string|string>: <optional>)` - A map of arbitrary string to string valued 
  user-provided metadata meant to describe the alias.

- `issuer` `(string: <optional>)` - Issuer URL of the external identity provider
  bound to this alias. Can be provided instead of `mount_accessor`. Cannot be

>   changed after creation.

Reviewer 2: ```suggestion
- `issuer` `(string: "")` - Issuer URL of the external identity provider
  bound to the alias. You cannot change `issuer` after creation. Setting
  `mount_accessor` is optional when you set `issuer`.
```
Style correction: avoid "this" as a pronoun
Format correction: For optional fields, provide the default rather than `<optional>`.

I'm assuming the two fields can coexist. If that's not true, we should change the language to indicate that the two fields are mutually exclusive.

- `external_id` `(string: <optional>)` - Subject identifier for the alias from the
  external identity provider. Must be unique with `issuer` within a namespace.

>   Cannot be changed after creation.

Reviewer 2: ```suggestion
- `external_id` `(string: "")` - Subject identifier for the alias from the
  external identity provider. The ID must be unique with `issuer` within a
  namespace. You cannot change `external_id` after creation.
```
Style correction: write in active voice, avoid implicit objects/subjects whenever possible

<Note>

For aliases bound to an OAuth resource-server configuration
(Enterprise), `mount_accessor` is a synthetic value with the format
`oauth-resource-server_<namespace_id>_<config_id>`, where `<config_id>` is the

> UUID of the configuration. Provide `issuer` instead and Vault generates it for you.

Reviewer 3: I would reword this with the issuer piece being first. Maybe something like:

```suggestion
For aliases bound to an OAuth resource-server configuration
(Enterprise), when only the `issuer` field is provided during creation, the `mount_accessor` will be an automatically generated synthetic value with the format
`oauth-resource-server_<namespace_id>_<config_id>`, where `<config_id>` is the
UUID of the matching configuration.
```

> </Note>

Reviewer 2: ```suggestion
```
Folded into relevant field info

- `custom_metadata` `(map<string|string>: <optional>)` - A map of arbitrary string to string valued 
  user-provided metadata meant to describe the alias.

- `issuer` `(string: "")` - Issuer URL bound to the alias. If provided, must

>   match the alias's existing `issuer`.

Reviewer 2: ```suggestion
- `issuer` `(string: "")` - Issuer URL bound to the alias. The provided issuer
  must match the existing `issuer` value for the alias.
```
Style correction: avoid possessives, use explicit subjects/objects

- `external_id` `(string: "")` - Subject identifier bound to the alias. If

>   provided, must match the alias's existing `external_id`.

Reviewer 2: ```suggestion
- `external_id` `(string: "")` - Subject identifier bound to the alias. The
  provided ID must match the existing `external_id` value for the alias.
```
Style correction: avoid possessives, use explicit subjects/objects
